gnomad_qc/v4/annotations/insilico_predictors.py

```diff
@@ -89,6 +100,94 @@ def _load_cadd_raw(cadd_tsv) -> hl.Table:
     return ht
 
 
+def create_revel_grch38_ht() -> hl.Table:
+    """
+    Create a Hail Table with REVEL scores for GRCh38.
+
+    .. note::
+    Starting with gnomAD v4, we use REVEL scores for only MANE Select and
+    canonical transcripts. Even when a variant falls on multiple MANE/canonical
+    transcripts of different genes, the scores are equal.
+    REVEL scores were downloaded from:
+       https://www.google.com/url?q=https%3A%2F%2Frothsj06.dmz.hpc.mssm.edu%2Frevel-v1.3_all_chromosomes.zip&sa=D&sntz=1&usg=AOvVaw2DS2TWUYl__0vqijzzxp5M
+       size ~648M, ~82,100,677 variants
+    REVEL's Ensembl ID is not from Ensembl 105, so we filter to transcripts
+    that are in Ensembl 105. The Ensembl 105 ID file was downloaded from Ensembl
+    105 archive. It contains the following columns:
+       Transcript stable ID, Ensembl Canonical, MANE Select
+    This deprecates the `has_duplicate` field present in gnomAD v3.
+
+    :return: Hail Table with REVEL scores for GRCh38.
+    """
+    revel_csv = "gs://gnomad-insilico/revel/revel-v1.3_all_chromosomes_with_transcript_ids.csv.bgz"
+    ensembl_ids = "gs://gnomad-insilico/ensembl/ensembl105id.tsv.bgz"
+
+    ht = hl.import_table(
+        revel_csv,
+        delimiter=",",
+        min_partitions=1000,
+        types={"grch38_pos": hl.tstr, "REVEL": hl.tfloat64},
+    )
+
+    logger.info("Annotating REVEL table...")
+    ht = ht.drop("hg19_pos", "aaref", "aaalt")
+    # drop variants that have no position in GRCh38 when lifted over from GRCh37
+    ht = ht.filter(ht.grch38_pos.contains("."), keep=False)
+    ht = ht.transmute(chr="chr" + ht.chr)
+    ht = ht.annotate(
+        locus=hl.locus(ht.chr, hl.int(ht.grch38_pos), reference_genome="GRCh38"),
+        alleles=hl.array([ht.ref, ht.alt]),
+        Transcript_stable_ID=ht.Ensembl_transcriptid.strip().split(";"),
+    )
+    ht = ht.select("locus", "alleles", "REVEL", "Transcript_stable_ID")
+    ht = ht.explode("Transcript_stable_ID")
+    ht = ht.key_by("Transcript_stable_ID")
+    print(ht.count())
+
+    logger.info("Import and process the ensembl ID file...")
+    ensembl_ids = hl.import_table(
+        ensembl_ids, min_partitions=200, impute=True, key="Transcript_stable_ID"
+    )
+    ensembl_ids = ensembl_ids.select("Ensembl_Canonical", "MANE_Select")
+    logger.info("Annotating revel HT with canonical and MANE Select transcripts...")
+    ht = ht.annotate(**ensembl_ids[ht.Transcript_stable_ID])
```

Comment:
Only commenting because below you use ht.key, isn't `Tanscript_stable_ID` the key? I'm fine with either way but it's helpful if references to keys are consistent for the person reading the code. Because you use key below to reference the same field, I end up asking myself why key wasnt used here and  revisit this piece to make sure I am understanding what is happening. So since these are interchangeable here, if I am understanding the code correctly, it would be helpful to either use only `key` or `Transcript _stable_ID` instead of flipping back and forth